### PR TITLE
added setRaiseOnException option

### PR DESCRIPTION
**Pull request by edwardmccaughan**
(I accidentally added the documentation and default value for @raise_on_exception in the other pull request, so they're not in this one... :-/ )

### CODE REVIEW

File: lib/routific.rb [hunk 1]
```diff
@@ -120,7 +121,11 @@ def job(job_id, token = @token)
                      method: :get,
                      token:  token
       if json
-        RoutificApi::Job.parse(json)
+        if @raise_on_exception && json["status"] == "error"
```
Comment (PragTob) [inline lib/routific.rb]: Just for my  better understanding I like to have the latter part here in () as I don't have to worry about the precedence of == versus &&

File: lib/routific.rb [hunk 2]
```diff
@@ -120,7 +121,11 @@ def job(job_id, token = @token)
...
+          raise ResponseError.new(json["output"])
```
Comment (PragTob) [inline lib/routific.rb]: why json["output"] and not just all of json?

Comment (PragTob) [inline lib/routific.rb]: seems good for now, we log all responses anyway and it is easy to adjust once we work with it

File: lib/routific.rb (hunk 1 above)
Comment (edwardmccaughan) [inline lib/routific.rb]: fixed in the newly pushed version

**Pull request merged**